### ISSUE TITLE
Build failing on Simulator with Xcode 12.0

### ISSUE THREAD

**Issue by sjsaul1101** [tanersener/react-native-ffmpeg#196]
**Description**
Builds fail on simulator when using Xcode 12.0. This might be related to different fields under the "Architecture" section in Build Settings.

**Expected behavior**
After following the installation steps (yarn add, pod install), the app was building successfully on Xcode 11.4. 

**Current behavior**
After upgrading Xcode, the build fails.

**Screenshots**
Architecture fields in Xcode 11.4:
<img width="702" alt="Screen Shot 2020-09-20 at 10 46 31 AM 1" src="https://user-images.githubusercontent.com/12602667/93718118-d3e09680-fb2e-11ea-99b7-ec30e5618e04.png">

Architecture fields in Xcode 12:
<img width="696" alt="Screen Shot 2020-09-20 at 10 47 52 AM" src="https://user-images.githubusercontent.com/12602667/93718117-cd521f00-fb2e-11ea-8ad9-f1a561929b50.png">

**Logs**
ld: in /Users/<USERNAME>/Desktop/<APPNAME>/ios/Pods/mobile-ffmpeg-https/libavcodec.framework/libavcodec(aacencdsp.o), building for iOS Simulator, but linking in object file built for free standing, file '/Users/<USERNAME>/Desktop/<APPNAME>/ios/Pods/mobile-ffmpeg-https/libavcodec.framework/libavcodec' for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)

**Environment**
 - Building after ejecting from Expo (sdk 37.0.3) and React Native v0.61.5

**Other**
I initially had a successful install, but it stopped working after Xcode/Simulator updated. I went back to an older version of Xcode and things seem to be working as expected.

**Comment by Salehjarad** [comment]
@sjsaul1101 <- _Did_ you found a solution .?

**Comment by sjsaul1101** [comment]
> @sjsaul1101 <- _Did_ you found a solution .?

I’ve been working around this by building with an older version of Xcode or, when using Xcode 12, it builds successful on a physical device.

**Comment by mohsinulhaq** [comment]
@tanersener any solution to this?

**Comment by tanersener** [comment]
I'm not sure about the reason. So, I don't have a fix for it right now. 

You can try using the version I'm working on from the `development` branch. It is still being developed but it does not have this issue.

**Comment by sjsaul1101** [comment]
> I'm not sure about the reason. So, I don't have a fix for it right now.
> 
> You can try using the new version I'm working on from the `development` branch. It is still being developed but it does work on Xcode 12.0.

**Comment by sjsaul1101** [comment]
Development branch fails in xCode 12, but with different errors:

<img width="491" alt="Screen Shot 2020-09-23 at 2 30 55 PM" src="https://user-images.githubusercontent.com/12602667/94072065-d6025980-fda9-11ea-9dab-2059c1710131.png">

**Comment by hb12DevTn** [comment]
To avoid cache issues I use the master version (0.4.4) and I apply a [patch](https://gist.github.com/hb12DevTn/8234049d0387dda802c8adb30ef882aa) using [patch-package](https://github.com/ds300/patch-package)

**Comment by tanersener** [comment]
I'm pushing changes to `development` every few days. Clean your `Cocoapods` cache and try again please.

```
pod cache clean react-native-ffmpeg --all
```

**Comment by dancixx** [comment]
Hi guys, i spent a lot of time with this issue, can i say days... I tried everything what i could. So my solution use react-native-ffmpeg@0.4.4 without patch, the patch for me was really bad more issue then before.  In node_modules/react-native-ffmpeg/react-native-ffmpeg.podspec file use 'https-lts' (default is the 'https'). I hope this helps you too. :)

**Comment by paramaggarwal** [comment]
I was able to unblock myself by adding this line before in Podfile:

```diff
target 'App' do
+ pod 'react-native-ffmpeg/min-lts', :path => '../node_modules/react-native-ffmpeg'

  config = use_native_modules!
  use_react_native!(:path => config["reactNativePath"])
```

Then run `pod install` once.

**Comment by Salehjarad** [comment]
### ☑︎ Here is how did i fix this to run my app on **"ios"**  simulator **"iPhone 11"**.
---

My versions in package.json
> `react-native-ffmpeg@0.4.4`
> `react-native@0.63.2`


**With the helps from :**
---> @tanersener 
---> @paramaggarwal 

Let's fix this issue in orders 👍 

---
- Cleaning `react-native-ffmpeg` pod caches
```zsh
   $=> cd ios
   $=> pod cache clean react-native-ffmpeg --all 
```
---
- Update your pod file in [ios] folder.
         -> Here is my pod file as example 😇
```js
require_relative '../node_modules/react-native/scripts/react_native_pods'
require_relative '../node_modules/@react-native-community/cli-platform-ios/native_modules'

platform :ios, '10.0'

target 'AppNameHere' do
  pod 'react-native-ffmpeg/full', :podspec => '../node_modules/react-native-ffmpeg/react-native-ffmpeg.podspec'
  pod 'react-native-ffmpeg/https-lts', :path => '../node_modules/react-native-ffmpeg'
  config = use_native_modules!

  use_react_native!(:path => config["reactNativePath"])

  target 'AppNameHere' do
    inherit! :complete
    # Pods for testing
  end

  # Enables Flipper.
  #
  # Note that if you have use_frameworks! enabled, Flipper will not work and
  # you should disable these next few lines.
  use_flipper!
  post_install do |installer|
    flipper_post_install(installer)
  end
end
```
---
- Installing Pod
```zsh
   $=> cd ios
   $=> pod install
```
   OR -> From your react native app root dir
```zsh
   $=> npx pod-install
``` 

---
- Cleaning your app build folder.
  Open your project in Xcode and press (⇧⌘K) to clean your build before.
 OR ->  in Xcode go to (Product => Clean build folder).

---
✅ Now run your app from Xcode, hopes its works like me.

---
==> Note: The error i had before was for "**ios**" simulator only, its works on my device as charm thanks to @tanersener .

**Comment by marcomarasco** [comment]
> Add this line in `Podfile`:
> 
> `pod 'react-native-ffmpeg/https-lts', :path => '../node_modules/react-native-ffmpeg'`
> 
> Disable auto-linking in `react-native.config.js` like this:
> 
> ```
> "react-native-ffmpeg": {
>   platforms: {
>     ios: null,
>   },
> },
> ```
> 
> Now run `pod install`

Disabling auto-linking is not necessary for me

**Comment by ami9000** [comment]
This is still an issue if we dont use LTS at least. I believe VLCKit also faced the same build issue for ios simulators due to Xcode 12 and libavcodec. Maybe this helps pin point a fix for us too. They were able to rebuild the library with a fix. https://code.videolan.org/videolan/VLCKit/-/issues/420

**Comment by tanersener** [comment]
As the maintainer of this project, I shared the situation with you and explained how you can use this plugin with Xcode 12.0.

`development` branch includes the new version of the plugin which works with Xcode 12.0. I'm working on it every day and push my changes regularly. It had minor compilation issues at some point last week but they're fixed immediately. It is stable and I'll release it soon. I'm working on the typescript definitions and the test-app right now. Functionality of the current test app is limited. Therefore I'm adding additional tabs to be able to test more features using it.

Thanks to everyone who provided instructions to use this plugin under Xcode 12. But, I provided a more official solution. And I accept that it's your choice whether to use a workaround (based on LTS version, a patch, etc) or an official solution. But I can't help you about the workarounds. So, I won't make any comments about them.

This is my final post about this plugin.

**Comment by tanersener** [comment]
Fixed in `v0.5.0`.
